packages/terra-folder-tree/src/FolderTree.jsx

```diff
@@ -54,6 +55,72 @@ const FolderTree = ({
   intl,
 }) => {
   const folderTreeID = `folder-tree-${uuidv4()}`;
+  const listNode = useRef();
+
+  const handleKeyDown = event => {
+    const listItems = listNode.current.querySelectorAll('[data-item-show-focus=true]');
+    // Remove all hidden list items
+    const visibleListItems = Array.prototype.slice.call(listItems).filter((item) => {
+      let parent = item.parentNode;
+      while (parent && parent !== listNode.current) {
+        if (parent.hasAttribute('hidden')) {
+          return false;
+        }
+        parent = parent.parentNode;
+      }
+      return true;
```

Comment:
Wouldn't it be easier to just remove hidden items from the DOM?  Then you wouldn't need this logic, right?

Reply:
I tried making these changes now, but it seems like it's not quite the easy fix that I thought it would be. The click logic that was already implemented needs all items to be present in the dom. We could address this in a later story.